c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -1455,7 +1455,7 @@ class PostgresCopyStreamWriter {
           NANOARROW_OK) {
         return ADBC_STATUS_INTERNAL;
       }
-      PostgresCopyFieldWriter* child_writer;
+      PostgresCopyFieldWriter* child_writer = {};
```

Comment:
I'm not sure why this is required; the reader doesn't need it, but without this the conda environment compiler was complaining

```
In file included from /usr/share/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/12.3.0/memory:75,
                 from /home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/statement.h:21,
                 from /home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/statement.cc:18:
In constructor 'std::__uniq_ptr_impl<_Tp, _Dp>::__uniq_ptr_impl(pointer) [with _Tp = adbcpq::PostgresCopyFieldWriter; _Dp = std::default_delete<adbcpq::PostgresCopyFieldWriter>]',
    inlined from 'std::__uniq_ptr_data<adbcpq::PostgresCopyFieldWriter, std::default_delete<adbcpq::PostgresCopyFieldWriter>, true, true>::__uniq_ptr_data(std::__uniq_ptr_impl<adbcpq::PostgresCopyFieldWriter, std::default_delete<adbcpq::PostgresCopyFieldWriter> >::pointer) [inherited from std::__uniq_ptr_impl<adbcpq::PostgresCopyFieldWriter, std::default_delete<adbcpq::PostgresCopyFieldWriter> >]' at /usr/share/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/12.3.0/bits/unique_ptr.h:233:40,
    inlined from 'std::unique_ptr<_Tp, _Dp>::unique_ptr(pointer) [with _Del = std::default_delete<adbcpq::PostgresCopyFieldWriter>; <template-parameter-2-2> = void; _Tp = adbcpq::PostgresCopyFieldWriter; _Dp = std::default_delete<adbcpq::PostgresCopyFieldWriter>]' at /usr/share/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/12.3.0/bits/unique_ptr.h:310:4,
    inlined from 'ArrowErrorCode adbcpq::PostgresCopyStreamWriter::InitFieldWriters(ArrowError*)' at /home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/postgres_copy_reader.h:1460:37,
    inlined from 'AdbcStatusCode adbcpq::{anonymous}::BindStream::ExecuteCopy(PGconn*, int64_t*, AdbcError*)' at /home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/statement.cc:554:7,
    inlined from 'AdbcStatusCode adbcpq::PostgresStatement::ExecuteUpdateBulk(int64_t*, AdbcError*)' at /home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/statement.cc:1225:3:
/usr/share/miniconda3/envs/test/x86_64-conda-linux-gnu/include/c++/12.3.0/bits/unique_ptr.h:168:56: error: 'child_writer' may be used uninitialized [-Werror=maybe-uninitialized]
  168 |       __uniq_ptr_impl(pointer __p) : _M_t() { _M_ptr() = __p; }
      |                                               ~~~~~~~~~^~~~~
In file included from /home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/statement.h:30:
/home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/postgres_copy_reader.h: In member function 'AdbcStatusCode adbcpq::PostgresStatement::ExecuteUpdateBulk(int64_t*, AdbcError*)':
/home/runner/work/arrow-adbc/arrow-adbc/c/driver/postgresql/postgres_copy_reader.h:1458:32: note: 'child_writer' was declared here
 1458 |       PostgresCopyFieldWriter* child_writer;
```

Intereresting enough both the conda version and my system gcc are 12.3.0

Reply:
Activating a conda environment also sets a bunch of other env vars, look at `cat $CONDA_PREFIX/etc/conda/activate.d/activate-gcc_linux-64.sh` inside a conda environment with compilers installed

Reply:
In particular

```
  CFLAGS_USED="-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem ${CONDA_PREFIX}/include"
  DEBUG_CFLAGS_USED="-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-all -fno-plt -Og -g -Wall -Wextra -fvar-tracking-assignments -ffunction-sections -pipe -isystem ${CONDA_PREFIX}/include"
  CPPFLAGS_USED="-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem ${CONDA_PREFIX}/include"
  DEBUG_CPPFLAGS_USED="-D_DEBUG -D_FORTIFY_SOURCE=2 -Og -isystem ${CONDA_PREFIX}/include"
  LDFLAGS_USED="-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,${CONDA_PREFIX}/lib -Wl,-rpath-link,${CONDA_PREFIX}/lib -L${CONDA_PREFIX}/lib"
  CMAKE_PREFIX_PATH_USED="${CONDA_PREFIX}:${CONDA_PREFIX}/x86_64-conda-linux-gnu/sysroot/usr"
```

Reply:
```suggestion
      PostgresCopyFieldWriter* child_writer = nullptr;
```
IMO this is a bit more idiomatic though

Reply:
Ah yea that makes sense. I always get tripped up over conda changing environment variables. One day I will remember...